### PR TITLE
Vlans and interfaces resource modules check and diff modes support

### PR DESCRIPTION
**Pull request by mingjunzhang2019**
SUMMARY
This is check and diff modes support for sonic_vlans and sonic_interfaces resource modules.

GitHub Issues
N/A

ISSUE TYPE
- Feature Pull Request

COMPONENT NAME
sonic_vlans, sonic_interfaces

OUTPUT
- vlans regression test result: 
[regression-2023-10-05-09-42-34.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12821687/regression-2023-10-05-09-42-34.html.pdf)
- vlans regression test log: 
[vlans_regression_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12821689/vlans_regression_test.log)
- vlans unit test script: 
[vlans.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12821693/vlans.txt)
- vlan unut test log: 
[vlans_unit_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12821694/vlans_unit_test.log)
- interfaces regression test result: 
[regression-2023-10-04-13-51-39.html.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12821707/regression-2023-10-04-13-51-39.html.pdf)
- interfaces regression test log: 
[Uploading interfaces_regression_test.log…]()
- interfaces unit test script: 
[interfaces.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12821750/interfaces.txt)
- interfaces unit test log: 
[interfaces_unit_test.log](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/12821752/interfaces_unit_test.log)

Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

How Has This Been Tested?

run regression tests and unit tests.

### CODE REVIEW

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
I am posting two comments (change requests), but am only requesting that one of them be made before merging this PR. The other one is retained in this PR for now, so it is available for reference in subsequent follow-up PRs needed soon for the "interfaces" resource module.

The change I am requesting now is just to add back a single instruction that is currently deleted in this change set when reading the currently configured interface speed.

Please let me know if there is a reason not to make the requested change.

File: plugins/module_utils/network/sonic/config/interfaces/interfaces.py [hunk 1]
```diff
@@ -341,17 +404,15 @@ def retrieve_default_intf_speed(self, intf_name):
                 self._module.fail_json(msg=str(exc), code=exc.code)
 
         # Read the speed
-        method = GET
+        intf_speed = 'SPEED_DEFAULT'
```
[the remark this answers is not in the document]
Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/interfaces/interfaces.py:407]: Done. TThanks.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
Thank you for making the requested change.

There are a few sanity errors that need to be fixed. (These weren't present in the previously posted version of this PR.) For the code coverage failures, it looks like two minor changes are needed in the UT "fixtures" file. The other two sanity failures can be eliminated by re-synching your fork and this PR branch to pick up the fix you recently provided to add a new Ansible core "ignore" version to the list of ignored module documentation checks for the non-existent "module" file corresponding to the enterprise_sonic "plugins/action/sonic.py" file. This fix was part of the change set for "Check and diff modes support for l2- and lag-interfaces modules" (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/303).

**Comment by mingjunzhang2019** [comment]
Fixed sanity error. Thanks.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
The code coverage (UT) failures were fixed by your most recent push today, but please also resync your fork and branch to verify that the other sanity failures (for "CI / sanity / py3.10 / ubuntu-latest / devel" and "CI / sanity / py3.11 / ubuntu-latest / devel") are fixed by the changes that were merged for "Check and diff modes support for l2- and lag-interfaces modules" (https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/303)".

**Comment by mingjunzhang2019** [comment]
Done the merge with main and fixed those sanity errors.

**Review by kerry-meyer — APPROVED** [review]
Thanks for doing the re-sync to verify the sanity results for this change set.

I see that all checks are passing now.

Approved.

**Pull request merged**